src/api/types.ts

```diff
@@ -0,0 +1,214 @@
+  tproxy_enabled: z.boolean(),
+  version: z.string().optional()
+});
+
+// Configuration Schema
+export const configurationSchema = z.object({
+  name: z.string(),
+  image: z.string(),
+  compose_file: composeFileSchema,
+  vcpu: z.number(),
+  memory: z.number(),
+  disk_size: z.number(),
+  ports: z.array(z.any())
+});
+
+// Hosted Schema
+export const hostedSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  status: z.string(),
+  uptime: z.string(),
+  app_url: z.string(),
+  app_id: z.string(),
+  instance_id: z.string(),
+  configuration: configurationSchema,
+  exited_at: z.string(),
+  boot_progress: z.string(),
+  boot_error: z.string(),
+  shutdown_progress: z.string(),
+  image_version: z.string()
+});
+
+// Managed User Schema
+export const managedUserSchema = z.object({
+  id: z.number(),
+  username: z.string()
+});
+
+// Node Schema
+export const nodeSchema = z.object({
+  id: z.number(),
+  name: z.string()
+});
+
+// CVM Instance Schema
+export const cvmInstanceSchema = z.object({
+  hosted: hostedSchema,
+  name: z.string(),
+  managed_user: managedUserSchema,
+  node: nodeSchema,
+  listed: z.boolean(),
+  status: z.string(),
+  in_progress: z.boolean(),
+  dapp_dashboard_url: z.string().nullable(),
+  syslog_endpoint: z.string(),
+  allow_upgrade: z.boolean()
+});
+
+// Create CVM Response Schema
+export const createCvmResponseSchema = z.object({
+  id: z.number(),
+  name: z.string(),
+  status: z.string(),
+  teepod_id: z.number(),
+  teepod: z.object({
+    id: z.number(),
+    name: z.string()
+  }),
+  user_id: z.number(),
+  app_id: z.string(),
+  vm_uuid: z.string().nullable(),
+  instance_id: z.string().nullable(),
+  app_url: z.string().nullable(),
+  base_image: z.string(),
+  vcpu: z.number(),
+  memory: z.number(),
+  disk_size: z.number(),
+  manifest_version: z.number(),
+  version: z.string(),
+  runner: z.string(),
+  docker_compose_file: z.string(),
+  features: z.array(z.string()).nullable(),
+  created_at: z.string(),
+  encrypted_env_pubkey: z.string()
+});
+
+// Get Pubkey From CVM Response Schema
+export const getPubkeyFromCvmResponseSchema = z.object({
+  app_env_encrypt_pubkey: z.string(),
+  app_id_salt: z.string()
+});
+
+// Get CVM By App ID Response Schema
+export const getCvmByAppIdResponseSchema = z.object({
+  id: z.number(),
+  teepod_id: z.number(),
+  teepod: z.object({
+    id: z.number(),
+    name: z.string()
+  }),
+  name: z.string(),
+  status: z.string(),
+  in_progress: z.boolean(),
+  app_id: z.string(),
+  vm_uuid: z.string(),
+  instance_id: z.string(),
+  vcpu: z.number(),
+  memory: z.number(),
+  disk_size: z.number(),
+  base_image: z.string(),
+  encrypted_env_pubkey: z.string(),
+  listed: z.boolean(),
+  project_id: z.string(),
+  project_type: z.string().nullable()
+});
+
+// Get User Info Response Schema
+export const getUserInfoResponseSchema = z.object({
+  username: z.string(),
+  email: z.string(),
+  credits: z.number(),
```

Comment:
For credit please use `ZodDecimal`, it can found here:
https://gist.github.com/shawncarr/a3819d4a7f98e8f185bc192e89b13de0

Usage:
```
balance: z.zodDecimal({ coerce: true })
```